Convert to latest Swift

rshevchuk opened the pull request.

indragiek commented on DominantColor/Shared/KMeans.swift:
```diff
@@ -71,7 +71,12 @@ func kmeans<T : ClusteredType>(
     
-    return map(Zip2(centroids, clusterSizes)) { Cluster(centroid: $0, size: $1) }
+        var clusters = [Cluster<T>]()
+        for (first, second) in Zip2Sequence(centroids, clusterSizes) {
```
Is there a reason why `Zip2Sequence.map` can't be used here instead of the for loop?

indragiek commented:
Looks good, just one comment to address. Thank you!

rshevchuk commented:
Thanks for the catch :) I found no reason there for the map replacement. Map can be used here. I fixed this and also added one fix for Swift 3 which is related to this https://github.com/apple/swift-evolution/blob/master/proposals/0004-remove-pre-post-inc-decrement.md

indragiek commented:
👍 Thank you very much!
